src/protocols/vnc/vnc.c

```diff
@@ -451,17 +451,21 @@ void* guac_vnc_client_thread(void* data) {
     guac_socket_flush(client->socket);
 
     guac_timestamp last_frame_end = guac_timestamp_current();
+    guac_timestamp server_runing_end = guac_timestamp_current();
```

Comment:
running*

Reply:
With something like the following:

```c
guac_timestamp last_frame_end = guac_timestamp_current();
guac_timestamp server_runing_end = guac_timestamp_current();
```

1. Repeatedly calling `guac_timestamp_current()` will incur unnecessary overhead. Unless there is an expectation that these values will be significantly different, it would be better to just reuse the value from the earlier call.
2. Can you clarify the difference between the use of these timestamps? Overall, they look very similar to the timestamps that are already being obtained for the frame timing heuristics.

Reply:
Okay so the difference between them is the frame ending requests are being operated in the do while loop. While this loop isn't gonna be called in idle state like wait message = 0. So, it will be operating in the following lines while waiting on the wait_result in line 462.
https://github.com/Noor-glyp/guacamole-server/blob/a4b98eac54fa168a43eca9b76d9094bf8739d16d/src/protocols/vnc/vnc.c#L457-L463